weave

```diff
@@ -918,6 +921,114 @@ proxy_addr() {
     return 1
 }
 
+######################################################################
+# launch helpers
+######################################################################
+
+launch_router() {
+    create_bridge
+    # We set the router name to the bridge mac since that is
+    # stable across re-creations of the containers.
+    #
+    # TODO use the mac of one of the physical host interfaces
+    # (eth0, wlan0, etc) so the name becomes stable across host
+    # restarts.
+    PEERNAME=$(cat /sys/class/net/$BRIDGE/address)
+    # backward compatibility...
+    if is_cidr "$1" ; then
+        echo "WARNING: $1 parameter ignored; 'weave launch' no longer takes a CIDR as the first parameter" >&2
+        shift 1
+    fi
+    CONTAINER_PORT=$PORT
+    ARGS=""
+    IPRANGE=
+    IPRANGE_SPECIFIED=
+    while [ $# -gt 0 ] ; do
+        case "$1" in
+            -password)
+                [ $# -gt 1 ] || usage
+                WEAVE_PASSWORD="$2"
+                export WEAVE_PASSWORD
+                shift 2
+                ;;
+            -port)
+                [ $# -gt 1 ] || usage
+                CONTAINER_PORT="$2"
+                shift 2
+                ;;
+            -iprange)
+                [ $# -gt 1 ] || usage
+                IPRANGE="$2"
+                IPRANGE_SPECIFIED=1
+                shift 2
+                ;;
+            *)
+                ARGS="$ARGS '$(echo "$1" | sed "s|'|'\"'\"'|g")'"
+                shift
+                ;;
+        esac
+    done
+    eval "set -- $ARGS"
+    if [ -z "$IPRANGE_SPECIFIED" ] ; then
+        IPRANGE="10.128.0.0/10"
+        if command_exists netcheck && ! netcheck $IPRANGE ; then
```

Comment:
It's intentional; if the user specifies a range then we leave checking it up to them.  Maybe they have a good reason for using a range that spans some existing route.